src/main/java/com/teamabnormals/abnormals_core/common/network/MessageS2CServerRedirect.java

```diff
@@ -0,0 +1,55 @@
+package com.teamabnormals.abnormals_core.common.network;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.screen.ConnectingScreen;
+import net.minecraft.client.gui.screen.DirtMessageScreen;
+import net.minecraft.client.gui.screen.MainMenuScreen;
+import net.minecraft.client.gui.screen.MultiplayerScreen;
+import net.minecraft.client.multiplayer.ServerData;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraftforge.fml.LogicalSide;
+import net.minecraftforge.fml.network.NetworkEvent;
+
+import java.util.function.Supplier;
+
+/**
+ * Allows servers to redirect clients to another server.
+ * @author Jackson
+ */
+public class MessageS2CServerRedirect {
+    private final String connectionAddress;
+
+    public MessageS2CServerRedirect(String address) {
+        this.connectionAddress = address;
+    }
+
+    public void serialize(PacketBuffer buf) {
+        buf.writeString(this.connectionAddress);
+    }
+
+    public static MessageS2CServerRedirect deserialize(PacketBuffer buf) {
+        return new MessageS2CServerRedirect(buf.readString());
+    }
+
+    public static void handle(MessageS2CServerRedirect message, Supplier<NetworkEvent.Context> ctx) {
+        NetworkEvent.Context context = ctx.get();
+        if (context.getDirection().getReceptionSide() == LogicalSide.CLIENT) {
+            context.enqueueWork(() -> {
+                Minecraft minecraft = Minecraft.getInstance();
```

Comment:
It'd be better to have the Minecraft instance be a constant field in this class instead of having to call `Minecraft.getInstance()` every time the packet is handled.